core/includes/actions.inc

```diff
@@ -73,16 +73,16 @@ function actions_get_info($action_name = NULL) {
  *
  * @param string $action_name
  *   The machine name of the action to be executed.
- * @param Entity $entity
+ * @param ?Entity|null $entity
```

Comment:
Did the IDE suggest this doc block notation?

Because PHPStan chokes on that. It accepts's either `@param Entity|null $entity` or `@param ?Entity $entity`, but not the combination of question mark and types separated by `|`.

Didn't find any official recommendation for nullable types in doc blocks, though.

Reply:
It was not suggested, I added it in some places because we has this `|null` notation in other places. I would be okay removing it from all places where the parameter is already indicated as nullable due to the question mark.

Reply:
```suggestion
 * @param ?Entity $entity
```

Reply:
Personally, I find it more readable if the types are fully written, like `@param Entity|null $entity`.
A single character's easily overlooked.
Adding `null` if it's nullable seems like a good idea. I belief, we have several misleading doc blocks in core re that.

But maybe that's just me - and I'm no doc block expert. :wink: Hopefully someone finds a good hint somewhere.

Reply:
The syntax everywhere should now be `* @param Entity|null $entity` in docblocks (no question mark, but `null` specified).